wearipedia/devices/oura/oura_ring3_gen.py

```diff
@@ -0,0 +1,344 @@
+    rest = random.randint(400, 600)
+    inactive = random.randint(500, 800)
+    low = random.randint(200, 400)
+    medium = random.randint(50, 150)
+    high = random.randint(0, 20)
+    inactivity_alerts = random.randint(0, 3)
+    average_met = round(random.uniform(1.0, 1.8), 2)
+
+    met_1min = [round(random.uniform(0.9, 5.0), 1) for _ in range(1440)]
+
+    activity_data = {
+        "summary_date": summary_date.strftime("%Y-%m-%d"),
+        "timezone": timezone_offset,
+        "day_start": start_time.isoformat(),
+        "day_end": end_time.isoformat(),
+        "cal_active": cal_active,
+        "cal_total": cal_total,
+        "class_5min": "".join(str(random.randint(0, 5)) for _ in range(288)),
+        "steps": steps,
+        "daily_movement": daily_movement,
+        "non_wear": non_wear,
+        "rest": rest,
+        "inactive": inactive,
+        "low": low,
+        "medium": medium,
+        "high": high,
+        "inactivity_alerts": inactivity_alerts,
+        "average_met": average_met,
+        "met_1min": met_1min,
+    }
+
+    return activity_data
+
+
+def get_readiness(date):
+
+    summary_date = datetime.strptime(date, "%Y-%m-%d")
+    score = random.randint(70, 100)
+    score_activity_balance = random.randint(70, 100)
+    score_hrv_balance = random.randint(70, 100)
+    score_previous_day = random.randint(70, 100)
+    score_previous_night = random.randint(70, 100)
+    score_recovery_index = random.randint(70, 100)
+    score_resting_hr = random.randint(70, 100)
+    score_sleep_balance = random.randint(70, 100)
+    score_temperature = random.randint(70, 100)
+
+    rest_mode_state = random.randint(0, 1)
+    period_id = random.randint(1, 4)
+
+    readiness_data = {
+        "summary_date": summary_date.strftime("%Y-%m-%d"),
+        "score": score,
+        "score_activity_balance": score_activity_balance,
+        "score_hrv_balance": score_hrv_balance,
+        "score_previous_day": score_previous_day,
+        "score_previous_night": score_previous_night,
+        "score_recovery_index": score_recovery_index,
+        "score_resting_hr": score_resting_hr,
+        "score_sleep_balance": score_sleep_balance,
+        "score_temperature": score_temperature,
+        "rest_mode_state": rest_mode_state,
+        "period_id": period_id,
+    }
+
+    return readiness_data
+
+
+def get_ideal_bedtime(date):
+    date = datetime.strptime(date, "%Y-%m-%d")
+
+    start_time = random.randint(0, 86400)
+    end_time = random.randint(start_time, 86400)
+
+    ideal_bedtime_data = {
+        "date": date.strftime("%Y-%m-%d"),
+        "bedtime_window": {"start": start_time, "end": end_time},
+        "status": "IDEAL_BEDTIME_AVAILABLE",
+    }
+
+    return ideal_bedtime_data
+
+
+def get_heart_rate(date):
+    date = datetime.strptime(date, "%Y-%m-%d")
+
+    timestamp = date.timestamp()
+
+    heart_rate_data = []
+
+    for i in range(0, 24 * 60, 5):
+        bpm = random.randint(50, 120)
```

Comment:
use a generator for this as opposed to random noise - i.e. Garmin Fenix 7s, Polar Verity Sense